scripts/_draw_info_display.py

```diff
@@ -0,0 +1,226 @@
+                     ].getboolean('Charges and Boosts')
+checkpoints_and_completion = config['INFO DISPLAY'
+                             ].getboolean('Checkpoints and Completion')
+miscellaneous = config['INFO DISPLAY'
+                ].getboolean('Miscellaneous')
+
+surface_properties = config['INFO DISPLAY'
+                     ].getboolean('Surface Properties')
+
+position = config['INFO DISPLAY'].getboolean('Position')
+
+stick = config['INFO DISPLAY'].getboolean('Stick')
+
+color = int(config['INFO DISPLAY']['Text Color'], 16)
+
+digits = config['INFO DISPLAY'].getint('Digits (to round to)')
+
+# draw information to the screen
+
+
+def create_infodisplay():
+    text = ""
+
+    if debug:
+        # test values here
+        text += f"{utils.get_game_id()}\n\n"
+    
+    if frame_count:
+        text += f"Frame: {core.get_frame_of_input()}\n\n"
+
+    if lap_splits:
+        # The actual max lap address does not update when crossing the finish line
+        # for the final time to finish the race. However, for whatever reason,
+        # race completion does. We use the "max" version to prevent lap times
+        # from disappearing when crossing the line backwards.
+        player_max_lap = math.floor(
+            classes.RaceInfoPlayer.race_completion_max())
+        lap_count = classes.RaceDataSettings.lap_count()
+
+        if player_max_lap >= 2 and lap_count > 1:
+            for lap in range(1, player_max_lap):
+                text += "Lap {:d}: {:02d}:{:012.9f}\n".format(
+                    lap, core.updateExactFinish(lap, 0)[0],
+                    core.updateExactFinish(lap, 0)[1])
+
+        if player_max_lap > lap_count:
+            text += "Final: {:02d}:{:012.9f}\n".format(
+                core.getUnroundedTime(lap_count, 0)[0],
+                core.getUnroundedTime(lap_count, 0)[1])
+        text += "\n"
+
+    if speed:
+        xz = core.get_speed().xz
+        y = core.get_speed().y
+        xyz = core.get_speed().xyz
+        engine_speed = classes.KartMove.speed()
+        cap = classes.KartMove.soft_speed_limit()
+        text += f"        XZ: {round(xz, digits)}\n"
+        text += f"       XYZ: {round(xyz, digits)}\n"
+        text += f"         Y: {round(y, digits)}\n"
+        text += f"    Engine: {round(engine_speed, digits)} / {round(cap, digits)}\n\n"
+
+    if internal_velocity:
+        iv_x = classes.VehiclePhysics.internal_velocity().x
+        iv_y = classes.VehiclePhysics.internal_velocity().y
+        iv_z = classes.VehiclePhysics.internal_velocity().z
+        text += f"      IV X: {round(iv_x, digits)}\n"
+        text += f"      IV Y: {round(iv_y, digits)}\n"
+        text += f"      IV Z: {round(iv_z, digits)}\n\n"
+
+    if internal_velocity_xyz:
+        iv_xyz = math.sqrt(classes.VehiclePhysics.internal_velocity().x ** 2
+               + classes.VehiclePhysics.internal_velocity().y ** 2 
+               + classes.VehiclePhysics.internal_velocity().z ** 2)
```

Comment:
This is 3 function calls, but is also a general operation we may apply to vec3 objects often. Instead of this, in mkw_classes.py, can you add a `norm()` function in the `vec3` class that computes what you're doing here? That way we can just call:

    iv_xyz = classes.VehiclePhysics.internal_velocity().norm()